Advanced-Recruit-Training/src/BatteryStateOfChargeService.h

```diff
@@ -25,4 +26,11 @@ class BatteryStateOfChargeService : public I_BatteryStateOfChargeService
     const double initialStateOfChargePercent_;
 
     // Add your necessary variables here
+    double current_;
+    double voltage_;
+    QTime time_;
+    double previousCurrent_;
+    QTime previousTime_;
```

Comment:
You don't need to track both the previous and and present current as member variables. I'll let you think about it.